src/Psl/Xml/Issue/Issue.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Xml\Issue;
+
+use LibXMLError;
+use Psl\Exception\InvariantViolationException;
+use Psl\Str;
+
+final class Issue
+{
+    private Level $level;
+    private int $code;
+    private int $column;
+    private string $message;
+    private string $file;
+    private int $line;
+
+    private function __construct(
+        Level $level,
+        int $code,
+        int $column,
+        string $message,
+        string $file,
+        int $line
+    ) {
+        $this->level = $level;
+        $this->code = $code;
+        $this->column = $column;
+        $this->message = $message;
+        $this->file = $file;
+        $this->line = $line;
+    }
+
+    /**
+     * @throws InvariantViolationException
+     */
+    public static function fromLibXmlError(LibXMLError $error): self
+    {
+        return new self(
+            Level::fromLevelCode($error->level),
+            $error->code,
+            $error->column,
+            $error->message,
+            $error->file,
+            $error->line,
+        );
+    }
+
```

Comment:
should be internal as well, again, we don't want to expose anything in the public API that reference the extensions used.

Reply:
I am really not sure what you are expecting from me here :)
It's just a named constructor on an *internal* class and it needs to be placed somewhere ...

Reply:
> an internal class

the class is not internal tho :p

Reply:
the function can be moved to `Psl\Xml\Internal\issue_from_xml_error` instead. and we keep this class exposed in the public API.

Reply:
ah ok that way, will try it later :)